connector/MongoDBAtlasODBC.pq

```diff
@@ -51,13 +52,71 @@ GetCredentialConnectionString = () as record =>
     in
         CredentialConnectionString;
 
-// SQL-1117: Implement ProcessUri Function
-ProcessUri = (uri as text) as record => 
+NullFieldCheck = (field as any, fieldName as text ) =>
+    let 
+        Result = if field <> null then Record.AddField([], fieldName, field) else []
+    in
+        Result;
+
+ProcessAuthSrc = (queryEntry as any, pathEntry as any) =>
     let
-        Result = []
+        DefaultAuthSrc = "admin",
+        FieldName = "AUTH_SRC",
+        // The 'authSource' connection option takes precedence over the /defaultauthdb value in the connection string.
+        // If both are unspecified, then authSource defaults to 'admin'.
+        AuthSrc =    if queryEntry <> null then
+                        queryEntry
+                    else if pathEntry <> null and Text.Length(Text.Remove(pathEntry, "/")) > 0 then
+                        Text.Remove(pathEntry, "/")
+                    else
+                        DefaultAuthSrc,
+        Result = Record.AddField([], FieldName, AuthSrc)
     in 
         Result;
 
+ProcessUri = (uri as text) as any => 
+    let
+        DefaultPort = "27017",
+        UriParts = Uri.Parts(uri),
+        BadUri = UriParts[Host] = "" or UriParts[Fragment] <> ""

```

Comment:
I am just wondering is extra parenthesis would help here.
Having 2 `=` following each others threw me off a bit. It looks like this is the expected syntax, so I'm thinking that if maybe it was written as `BadUri = (UriParts[Host] = "") or (UriParts[Fragment] <> "") ... it would help a little bit. Or maybe just having each condition on its own line. I don't know. Just not a big fan of this.